### PR TITLE
add rescale test

### PR DESCRIPTION
**Pull request by sed-i**
Add a test for scale up/down and trigger leadership change on the way.

### CODE REVIEW

**Review by simskij — APPROVED** [review]

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -79,3 +80,21 @@ async def __aenter__(self):
...
+
+async def block_until_leader_elected(ops_test: OpsTest, app_name: str):
+    async def is_leader_elected():
+        units = ops_test.model.applications[app_name].units
+        return any([await units[i].is_leader_from_status() for i in range(len(units))])
+
+    # await ops_test.model.block_until(is_leader_elected)
+    # block_until does not take async (yet?) https://github.com/juju/python-libjuju/issues/609
+    while not await is_leader_elected():
+        await asyncio.sleep(5)
```
Comment (Abuelodelanada) [inline tests/integration/helpers.py:100]: Why we are using inner function here?

File: tests/integration/test_rescale_charm.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import pytest
+import yaml
+from helpers import (  # type: ignore[attr-defined]
+    IPAddressWorkaround,
+    block_until_leader_elected,
+    get_leader_unit_num,
+    get_unit_address,
+)
+
+log = logging.getLogger(__name__)
```
Comment (Abuelodelanada) [inline tests/integration/test_rescale_charm.py:28]: Why not `logger` as usual?:

```python
logger = logging.getLogger(__name__)
```

**Review by Abuelodelanada — APPROVED** [review]

File: tests/integration/helpers.py (hunk 1 above)
Comment (simskij) [reply tests/integration/helpers.py:100]: good catch! i thought we agreed not to do that :)

Comment (sed-i) [reply tests/integration/helpers.py]: Done

File: tests/integration/test_rescale_charm.py (hunk 2 above)
Comment (sed-i) [reply tests/integration/test_rescale_charm.py]: Done

**Pull request merged**